hera_qm/uvflag.py

```diff
@@ -0,0 +1,591 @@
+
+                self.type = header['type'].value
+                self.mode = header['mode'].value
+                self.time_array = header['time_array'].value
+                self.lst_array = header['lst_array'].value
+                self.freq_array = header['freq_array'].value
+                self.history = header['history'].value + ' Read by ' + hera_qm_version_str
+                self.history += history
+                if 'label' in header.keys():
+                    self.label = header['label'].value
+                self.polarization_array = header['polarization_array'].value
+                if self.type == 'baseline':
+                    self.baseline_array = header['baseline_array'].value
+                    self.ant_1_array = header['ant_1_array'].value
+                    self.ant_2_array = header['ant_2_array'].value
+                elif self.type == 'antenna':
+                    self.ant_array = header['ant_array'].value
+
+                dgrp = f['/Data']
+                if self.mode == 'metric':
+                    self.metric_array = dgrp['metric_array'].value
+                elif self.mode == 'flag':
+                    self.flag_array = dgrp['flag_array'].value
+
+                self.weights_array = dgrp['weights_array'].value
+
+            self.clear_unused_attributes()
+
+    def write(self, filename, clobber=False, data_compression='lzf'):
+        """
+        Write a UVFlag object to a hdf5 file.
+
+        Args:
+            filename: The file to write to.
+            clobber: Option to overwrite the file if it already exists. Default is False.
+            data_compression: HDF5 filter to apply when writing the data_array. Default is
+                 LZF. If no compression is wanted, set to None.
+        """
+
+        if os.path.exists(filename):
+            if clobber:
+                print('File ' + filename + ' exists; clobbering')
+            else:
+                raise ValueError('File ' + filename + ' exists; skipping')
+
+        with h5py.File(filename, 'w') as f:
+            header = f.create_group('Header')
+
+            # write out metadata
+            header['type'] = self.type
+            header['mode'] = self.mode
+            header['time_array'] = self.time_array
+            header['lst_array'] = self.lst_array
+            header['freq_array'] = self.freq_array
+            header['polarization_array'] = self.polarization_array
+            header['history'] = self.history + 'Written by ' + hera_qm_version_str
+            header['label'] = self.label
+
+            if self.type == 'baseline':
+                header['baseline_array'] = self.baseline_array
+                header['ant_1_array'] = self.ant_1_array
+                header['ant_2_array'] = self.ant_2_array
+            elif self.type == 'antenna':
+                header['ant_array'] = self.ant_array
+
+            dgrp = f.create_group("Data")
+            if data_compression is not None:
```

Comment:
the h5py.File.create_dataset function takes compression=None as default in its source code, we could collapse this if/else statement

Reply:
done.